### ISSUE TITLE
Installation on Python 3.11 fails, when Doxygen is installed

### ISSUE THREAD

**Issue by amotl** [uber/h3-py#298]
Hi there,

first things first: Thanks a stack for conceiving and maintaining this excellent library.

Recognizing that there are no `cp311` wheels available on PyPI for this package yet, we expected that installation on Python 3.11 would need a compiler to build a binary distribution before installing it.

Normally, it works perfectly well. However, we discovered that the build breaks when Doxygen is installed on the system. The corresponding error output is:

```
  -- Found Doxygen: /usr/bin/doxygen (found version "1.9.1") found components: doxygen dot
  CMake Error: File /tmp/pip-install-m2znrsp5/h3_c64c4a4f63ec4918ad0bc7f40418f81d/src/h3lib/dev-docs/Doxyfile.in does not exist.
  CMake Error at src/h3lib/CMakeLists.txt:299 (configure_file):
    configure_file Problem configuring file
```

Indeed, there is no directory `dev-docs` within `/tmp/pip-install-m2znrsp5/h3_c64c4a4f63ec4918ad0bc7f40418f81d/src/h3lib`.

When Doxygen is not installed on the system, the CMake build procedure would not invoke that target at all, so building and installing succeeds flawlessly.

With kind regards,
Andreas.

### Reproduction

```
docker run --rm -it python:3.11-slim-bullseye bash
apt-get update
apt-get install --yes build-essential doxygen
pip install h3 --no-clean --verbose
```

**Comment by ajfriend** [comment]
This should be mitigated when we have 3.11 wheels up, but ideally, we'd avoid this problem all together so folks could build from source when needed.

@isaacbrodsky Can we configure CMake to avoid Doxygen when building `h3-py`?

**Comment by ajfriend** [comment]
I'm surprised we're getting this issue since we set:

https://github.com/uber/h3-py/blob/3eece401ba311feeba99073eb65db2764ef85557/CMakeLists.txt#L22

**Comment by isaacbrodsky** [comment]
What version of h3-py (and of H3) is being installed? According to the upstream CMakeLists.txt it should not try to configure that file if ENABLE_DOCS is OFF. Is it possible the version being installed does not correspond to the CMakeLists.txt you linked to?

**Comment by ajfriend** [comment]
Yup, that must be it. I don't think we had the CMake changes in ther at the time of 3.7.4: https://github.com/uber/h3-py/blob/d1a5a3b9b2add6f970a840f154c51947d312cdd7/CMakeLists.txt

**Comment by ajfriend** [comment]
Either of these releases should work:

- https://github.com/uber/h3-py/releases/tag/v3.7.6
- https://github.com/uber/h3-py/releases/tag/v4.0.0b2

@amotl, let me know if you have any issues. (If you're up for it, you could even try `pip install h3 --no-binary :all:` to check that it works without the wheel.)

**Comment by amotl** [comment]
Dear @ajfriend,

I have been able to confirm that installing the version 3.7.6 sdist package works well.
```
pip install --verbose git+https://github.com/uber/h3-py@v3.7.6
```

However, I also just discovered that version 3.7.6 wheels have now been uploaded to PyPI, see https://pypi.org/project/h3/3.7.6/. Installing from PyPI also works without issues now, and does not need a compiler. Excellent!

```
docker run --rm -it python:3.11-slim-bullseye bash
pip install h3
python -c "import h3; print(h3.__version__)"
```

```
3.7.6
```

Thank you very much.

With kind regards,
Andreas.
